EIPS/eip-5450.md

```diff
@@ -13,51 +13,108 @@ requires: 3540, 3670, 4200, 4750
 
 ## Abstract
 
-Introduce extended validation of code sections to guarantee that neither stack underflow nor overflow can happen during execution of validated contracts.
+Introduce extended validation of EOF code sections to guarantee that neither stack underflow nor overflow can happen during execution of validated contracts.
 
 ## Motivation
 
-Currently existing EVM implementations perform a number of validity checks for each executed instruction, such as check for stack overflow/underflow, sufficient gas, etc. This change aims to minimize the number of such checks required at run-time, by verifying at deploy-time that no exceptional conditions can happen, and preventing to deploy the code where it could happen.
+The current EVM performs a number of validity checks for each executed instruction, such as checking
+for instruction being defined, stack overflow and underflow, and enough amount of gas remaining.
 
-In particular, this extended code validation eliminates the need for EVM stack underflow checks done for every executed instruction. It also prevents deploying code that can be statically proven to require more than 1024 stack items and limits stack overflow checks to the `CALLF` instruction only.
+This EIP minimizes the number of such checks required at run-time
+by verifying that no exceptional conditions can happen
+and preventing the execution and deployment of any invalid code.
+
+The operand stack validation provides several benefits:
+
+- removes run-time stack underflow check for all instructions,
+- removes run-time stack overflow check for all instruction except `CALLF`,
+- ensures that an execution terminates with one of the terminating instructions,
+- prevents the deployment of code with unreachable instructions, which discourages the use of code sections for data storage.
+
+It also has some disadvantages:
+
+- adds constraints to the code structure (similar to JVM, CPython bytecode, WebAssembly and others); however, these constraints can be lifted in a backward-compatible manner if they are shown to be user-unfriendly,
+- adds second validation pass; however, validation's computational and space complexity remains linear.
 
 ## Specification
 
 ### Code validation
 
-*Remark:* We rely on the notions of *data stack* and *type section* as defined by [EIP-4750](./eip-4750.md).
+*Remark:* We rely on the notions of *operand stack* (FIXME) and *type section* as defined by [EIP-4750](./eip-4750.md).
```

Comment:
still a FIXME here

Reply:
This should stay: we want to rename "data stack" to "operand stack" across all EIPs.